porch/controllers/packagevariants/pkg/controllers/packagevariant/packagevariant_controller.go

```diff
@@ -713,3 +720,34 @@ func ensurePackageContext(pv *api.PackageVariant,
 	prr.Spec.Resources["package-context.yaml"] = cm.String()
 	return nil
 }
+
+func ensureKRMFunctions(pv *api.PackageVariant,
+	prr *porchapi.PackageRevisionResources) error {
+
+	if len(pv.Spec.Mutators) == 0 {
+		return nil
+	}
+
+	if _, ok := prr.Spec.Resources[kptfilev1.KptFileName]; !ok {
+		return fmt.Errorf("%s not found in PackageRevisionResources '%s/%s'", kptfilev1.KptFileName, prr.Namespace, prr.Name)
+	}
+
+	kptfile, err := internalpkg.DecodeKptfile(strings.NewReader(prr.Spec.Resources[kptfilev1.KptFileName]))
+	if err != nil {
+		return fmt.Errorf("failed to parse %s of PackageRevisionResources '%s/%s'", kptfilev1.KptFileName, prr.Namespace, prr.Name)
+	}
+
+	// TODO q: why are PackageVariant mutators being run before the ones in the kptfile?
+	// Wouldn't that make it harder for the vendors of the the original package, because their mutators now need
+	// to account for the possibility of mutators from end-users?
```

Comment:
Originally, I considered having the design allowing both prefix and suffix lists of functions, but I didn't see a real use case for it yet. Besides, the function pipeline needs to run and be idempotent. I chose prefix because it allows the downstream author to tweak what gets fed into the original pipeline, though that's somewhat arbitrary. If we have prefix working, suffix is easy to add later.

The original package authors should never make assumptions about the pipeline not having additional items. Users can change it after cloning the package, this is normal usage.